api/src/main/java/com/velocitypowered/api/event/player/KickedFromServerEvent.java

```diff
@@ -20,7 +21,7 @@
 
   private final Player player;
   private final RegisteredServer server;
-  private final @Nullable Component originalReason;
+  private final net.kyori.adventure.text.Component originalReason;
```

Comment:
Is the `@Nullable` annotation removal both here and in the new constructor intentional?
If so, the constructor is missing the non-null precondition.